servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContextHolderThread.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2018-2019, 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import javax.annotation.Nullable;
+
+class AsyncContextHolderThread extends Thread implements AsyncContextMapHolder {
+    @Nullable
+    private AsyncContextMap asyncContextMap;
```

Comment:
It makes me feel itchy that this field is not volatile.

Reply:
If it is only ever set/read on the thread for itself then agreed there is no issue.